WeakAuras/WeakAuras.lua

```diff
@@ -2288,65 +2287,69 @@ local function validateUserConfig(data, options, config)
       end
     end
   end
-  for i = #options, 1, -1 do
```

Comment:
turns out if this code ever did remove an option (except for possibly the last one), then the following block would immediately error since the `authorOptionKeys` table doesn't make any sense. So we just delay the removal of garbage entries in this array to the end of the function.